File: default/src/app/_models/order.ts
import { User } from './user';
import { OrderDetail } from './orderdetail';

export interface Order {
    id: number;
    userId: number;
    createdAt: Date;
    status: number;
    user: User;
    orderDetail: OrderDetail[];
    codeId: number;
    total: number;
}

File: Learning.API/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Learning.API.Models
{
    public class Order
    {
        public int Id { get; set; }
        public User User { get; set; }
        public int UserId { get; set; }
        public DateTime CreatedAt { get; set; }
        public int Status { get; set; }
        public ICollection<OrderDetail> OrderDetail { get; set; }

        public int Total { get; set; }


        public int? CodeId { get; set; }

    }
}